util/argo/argo.go

```diff
@@ -714,11 +711,10 @@ func verifyGenerateManifests(
 	repositoryCredentials []*argoappv1.RepoCreds,
 	enableGenerateManifests map[string]bool,
 	settingsMgr *settings.SettingsManager,
-	hasMultipleSources bool,
```

Comment:
Why would we remove this?

Reply:
Cause we are using Application as argument instead of Application.Name, Application.Spec.HasMultipleSources etc,.

Related with the previous comment, I haven't strong opinion on that then if you prefer, I can undo this change and add to the list of arguments the value of the Annotation.